droidlet/tools/hitl/utils/job_management.py

```diff
@@ -0,0 +1,296 @@
+
+# for ecr
+AWS_ECR_ACCESS_KEY_ID = os.environ["AWS_ECR_ACCESS_KEY_ID"]
+AWS_ECR_SECRET_ACCESS_KEY = os.environ["AWS_ECR_SECRET_ACCESS_KEY"]
+AWS_ECR_REGION = os.environ["AWS_ECR_REGION"] if os.getenv("AWS_ECR_REGION") else "us-west-1"
+AWS_ECR_REGISTRY_ID = "492338101900"
+AWS_ECR_REPO_NAME = "craftassist"
+
+# local tmp directory
+HITL_TMP_DIR = (
+    os.environ["HITL_TMP_DIR"] if os.getenv("HITL_TMP_DIR") else f"{os.path.expanduser('~')}/.hitl"
+)
+
+# job management record path prefix
+JOB_MNG_PATH_PREFIX = "job_management_records"
+
+ecr = boto3.client(
+    "ecr",
+    aws_access_key_id=AWS_ECR_ACCESS_KEY_ID,
+    aws_secret_access_key=AWS_ECR_SECRET_ACCESS_KEY,
+    region_name=AWS_ECR_REGION,
+)
+
+s3 = boto3.resource(
+    "s3",
+    region_name=AWS_DEFAULT_REGION,
+    aws_access_key_id=AWS_ACCESS_KEY_ID,
+    aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
+)
+
+bucket = s3.Bucket(S3_BUCKET_NAME)
+
+
+class MetaData(Enum):
+    BATCH_ID = "batch_id"
+    NAME = "name"
+    S3_LINK = "s3_link"
+    COST = "cost"
+    START_TIME = "start_time"
+    END_TIME = "end_time"
+
+
+class Job(Enum):
+    INTERACTION = "interaction"
+    ANNOTATION = "annotation"
+    RETRAIN = "retrain"
+
+
+class JobStat(Enum):
+    ENABLED = "enabled"
+    NUM_REQUESTED = "num_requested"
+    NUM_COMPLETED = "num_completed"
+    START_TIME = "start_time"
+    END_TIME = "end_time"
+    NUM_SESSION_LOG = "num_session_log"
+    NUM_COMMAND = "num_command"
+    NUM_ERR_COMMAND = "num_err_command"
+    DASHBOARD_VER = "dashboard_ver"
+    ORI_DATA_SZ = "ori_data_sz"
+    NEW_DATA_SZ = "new_data_sz"
+    MODEL_ACCURACY = "model_accuracy"
+    MODEL_EPOCH = "model_epoch"
+    MODEL_LOSS = "model_loss"
+
+
+# statastics that all jobs have
+STAT_FOR_ALL = set(
+    [
+        JobStat.ENABLED,
+        JobStat.NUM_REQUESTED,
+        JobStat.NUM_COMPLETED,
+        JobStat.START_TIME,
+        JobStat.END_TIME,
+    ]
+)
+
+# statatics that are unique for a job (not in the STAT_FOR_ALL set)
+STAT_JOB_PAIR = {
+    Job.INTERACTION: set(
+        [
+            JobStat.NUM_SESSION_LOG,
+            JobStat.NUM_COMMAND,
+            JobStat.NUM_ERR_COMMAND,
+            JobStat.DASHBOARD_VER,
+        ]
+    ),
+    Job.RETRAIN: set(
+        [
+            JobStat.ORI_DATA_SZ,
+            JobStat.NEW_DATA_SZ,
+            JobStat.MODEL_ACCURACY,
+            JobStat.MODEL_EPOCH,
+            JobStat.MODEL_LOSS,
+        ]
+    ),
+}
+
+# update job stat interval in seconds
+DEFAULT_STAT_UPDATE_INTERVAL = 60
+
+
+def get_dashboard_version(image_tag: str):
+    """
+    get sha256 value of the input image_tag
+    """
+    response = ecr.batch_get_image(
+        registryId=AWS_ECR_REGISTRY_ID,
+        repositoryName=AWS_ECR_REPO_NAME,
+        imageIds=[
+            {"imageTag": image_tag},
+        ],
+    )
+    assert len(response["images"]) == 1
+    return response["images"][0]["imageId"]["imageDigest"]
+
+
+def get_s3_link(batch_id: int):
```

Comment:
This file should only contains the pure abstraction class here. Functions like `get_s3_link` should be moved to a separate util file.